store/feature-store/src/main/java/org/calrissian/accumulorecipes/featurestore/FeatureStore.java

```diff
@@ -50,4 +50,33 @@
      */
     <T extends Feature> CloseableIterable<T> query(Date start, Date end, String group, Set<String> types,
                                                    String name, TimeUnit timeUnit, Class<T> featureType, Auths auths);
+
+    /**
+     * Returns all unique groups from the index. Only groups starting with the given prefix will be returned.
+     * An empty prefix will return all groups.
+     * @param prefix
+     * @param auths
+     * @return
+     */
+    Iterable<String> groups(String prefix, Auths auths);
+
+    /**
+     * Returns all unique types from the index for a given group. Only types starting with the given prefix
+     * will be returned. An empty prefix will return all types in the group.
+     * @param group
+     * @param prefix
+     * @param auths
+     * @return
+     */
+    Iterable<String> types(String group, String prefix, Auths auths);
+
+    /**
+     * Returns all unique names for a given group and type. Only names starting with the given prefix
+     * will be returned. An empty prefix will return all names for the given group and type.
+     * @param typeAndGroup
+     * @param prefix
+     * @param auths
+     * @return
+     */
+    Iterable<String> names(Pair<String,String> typeAndGroup, String prefix, Auths auths);
```

Comment:
Good point, I originally had this as an Iterable and overlooked it when I removed it